src/platform/plugins/shared/dashboard/public/services/dashboard_backup_service.ts

```diff
@@ -130,31 +132,22 @@ class DashboardBackupService implements DashboardBackupServiceType {
 
   public getDashboardIdsWithUnsavedChanges() {
     try {
-      const dashboardStatesInSpace =
-        this.sessionStorage.get(DASHBOARD_STATE_SESSION_KEY)?.[this.activeSpaceId] ?? {};
-      const dashboardsSet: Set<string> = new Set<string>();
-
-      Object.keys(dashboardStatesInSpace).map((dashboardId) => {
-        if (
-          dashboardStatesInSpace[dashboardId].viewMode === 'edit' &&
-          Object.keys(dashboardStatesInSpace[dashboardId]).some(
-            (stateKey) => stateKey !== 'viewMode' && stateKey !== 'references'
```

Comment:
You've removed the `references` check from here too. Have you verified that this is not needed? Maybe by ensuring that the behaviours that make a Dashboard show up in the unsaved listing are identical between this branch and main?